Demo script does not work

**Describe the bug**
The demo script fails with `TypeError: NA.__float__ returned non-float (type NA)`

**To Reproduce**
Steps to reproduce the behavior:
1. Install pinecore via pip
2. Download some data (`pyne data download ccxt --symbol "BYBIT:BTC/USDT:USDT"`)
3. (Optional) add `module_properties.json` (until solved #1)
4. Run autogenerated `demo.py` script 
 
**Operating System (please complete the following information):**
 - OS: Linux

**Additional context**
The script I run
```python
"""
@pyne
Simple Pyne Script Demo

A basic demo showing a 12 and 26 period EMA crossover system.
"""
from pynecore import Series
from pynecore.lib import script, input, plot, color, ta


@script.indicator(
    title="Simple EMA Crossover Demo",
    shorttitle="EMA Demo",
    overlay=True
)
def main(
    src: Series[float] = input.source("close", title="Price Source"),
    fast_length: int = input.int(12, title="Fast EMA Length"),
    slow_length: int = input.int(26, title="Slow EMA Length")
):
    """
    A simple EMA crossover demo
    """
    # Calculate EMAs
    fast_ema = ta.ema(src, fast_length)
    slow_ema = ta.ema(src, slow_length)

    # Plot our indicators
    plot(fast_ema, title="Fast EMA", color=color.blue)
    plot(slow_ema, title="Slow EMA", color=color.red)
```

```
╭────────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────────────────────────────────────╮
│ /home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/cli/commands/run.py:142 in run                                                                      │
│                                                                                                                                                                                       │
│   139 │   │   │   runner = ScriptRunner(script, ohlcv_iter, syminfo, last_bar_index=size - 1,                                                                                         │
│   140 │   │   │   │   │   │   │   │     plot_path=plot_path, strat_path=strat_path, equity_pat                                                                                        │
│   141 │   │   │   # Run the script                                                                                                                                                    │
│ ❱ 142 │   │   │   runner.run(on_progress=cb_progress)                                                                                                                                 │
│   143                                                                                                                                                                                 │
│                                                                                                                                                                                       │
│ ╭────────────────────────────────────────────────── locals ──────────────────────────────────────────────────╮                                                                        │
│ │          data = PosixPath('/home/kpob/workspace/trading/pine/workdir/data/ccxt_BINANCE_BTC_USDT_1D.ohlcv') │                                                                        │
│ │   equity_path = PosixPath('/home/kpob/workspace/trading/pine/workdir/output/demo_equity.csv')              │                                                                        │
│ │    ohlcv_iter = <generator object OHLCVReader.read_from at 0x70774ed25340>                                 │                                                                        │
│ │     plot_path = PosixPath('/home/kpob/workspace/trading/pine/workdir/output/demo.csv')                     │                                                                        │
│ │      progress = <rich.progress.Progress object at 0x70774e97a7b0>                                          │                                                                        │
│ │        reader = <pynecore.core.ohlcv_file.OHLCVReader object at 0x70774e9b7e70>                            │                                                                        │
│ │        runner = <pynecore.core.script_runner.ScriptRunner object at 0x70774e9d5bc0>                        │                                                                        │
│ │        script = PosixPath('/home/kpob/workspace/trading/pine/workdir/scripts/demo.py')                     │                                                                        │
│ │          size = 364                                                                                        │                                                                        │
│ │    strat_path = PosixPath('/home/kpob/workspace/trading/pine/workdir/output/demo_strat.csv')               │                                                                        │
│ │       syminfo = SymInfo(                                                                                   │                                                                        │
│ │                 │   prefix='BINANCE',                                                                      │                                                                        │
│ │                 │   description='BTC / USDT Spot',                                                         │                                                                        │
│ │                 │   ticker='BTCUSDT',                                                                      │                                                                        │
│ │                 │   currency='USDT',                                                                       │                                                                        │
│ │                 │   basecurrency='BTC',                                                                    │                                                                        │
│ │                 │   period='1D',                                                                           │                                                                        │
│ │                 │   type='crypto',                                                                         │                                                                        │
│ │                 │   volumetype='base',                                                                     │                                                                        │
│ │                 │   mintick=0.01,                                                                          │                                                                        │
│ │                 │   pricescale=100,                                                                        │                                                                        │
│ │                 │   minmove=1.0,                                                                           │                                                                        │
│ │                 │   pointvalue=1.0,                                                                        │                                                                        │
│ │                 │   opening_hours=[                                                                        │                                                                        │
│ │                 │   │   SymInfoInterval(day=0, start=datetime.time(0, 0), end=datetime.time(23, 59, 59)),  │                                                                        │
│ │                 │   │   SymInfoInterval(day=1, start=datetime.time(0, 0), end=datetime.time(23, 59, 59)),  │                                                                        │
│ │                 │   │   SymInfoInterval(day=2, start=datetime.time(0, 0), end=datetime.time(23, 59, 59)),  │                                                                        │
│ │                 │   │   SymInfoInterval(day=3, start=datetime.time(0, 0), end=datetime.time(23, 59, 59)),  │                                                                        │
│ │                 │   │   SymInfoInterval(day=4, start=datetime.time(0, 0), end=datetime.time(23, 59, 59)),  │                                                                        │
│ │                 │   │   SymInfoInterval(day=5, start=datetime.time(0, 0), end=datetime.time(23, 59, 59)),  │                                                                        │
│ │                 │   │   SymInfoInterval(day=6, start=datetime.time(0, 0), end=datetime.time(23, 59, 59))   │                                                                        │
│ │                 │   ],                                                                                     │                                                                        │
│ │                 │   session_starts=[                                                                       │                                                                        │
│ │                 │   │   SymInfoSession(day=0, time=datetime.time(0, 0)),                                   │                                                                        │
│ │                 │   │   SymInfoSession(day=1, time=datetime.time(0, 0)),                                   │                                                                        │
│ │                 │   │   SymInfoSession(day=2, time=datetime.time(0, 0)),                                   │                                                                        │
│ │                 │   │   SymInfoSession(day=3, time=datetime.time(0, 0)),                                   │                                                                        │
│ │                 │   │   SymInfoSession(day=4, time=datetime.time(0, 0)),                                   │                                                                        │
│ │                 │   │   SymInfoSession(day=5, time=datetime.time(0, 0)),                                   │                                                                        │
│ │                 │   │   SymInfoSession(day=6, time=datetime.time(0, 0))                                    │                                                                        │
│ │                 │   ],                                                                                     │                                                                        │
│ │                 │   session_ends=[                                                                         │                                                                        │
│ │                 │   │   SymInfoSession(day=0, time=datetime.time(23, 59, 59)),                             │                                                                        │
│ │                 │   │   SymInfoSession(day=1, time=datetime.time(23, 59, 59)),                             │                                                                        │
│ │                 │   │   SymInfoSession(day=2, time=datetime.time(23, 59, 59)),                             │                                                                        │
│ │                 │   │   SymInfoSession(day=3, time=datetime.time(23, 59, 59)),                             │                                                                        │
│ │                 │   │   SymInfoSession(day=4, time=datetime.time(23, 59, 59)),                             │                                                                        │
│ │                 │   │   SymInfoSession(day=5, time=datetime.time(23, 59, 59)),                             │                                                                        │
│ │                 │   │   SymInfoSession(day=6, time=datetime.time(23, 59, 59))                              │                                                                        │
│ │                 │   ],                                                                                     │                                                                        │
│ │                 │   timezone='UTC',                                                                        │                                                                        │
│ │                 │   avg_spread=None,                                                                       │                                                                        │
│ │                 │   taker_fee=0.001,                                                                       │                                                                        │
│ │                 │   maker_fee=0.001                                                                        │                                                                        │
│ │                 )                                                                                          │                                                                        │
│ │          task = 0                                                                                          │                                                                        │
│ │     time_from = datetime.datetime(2024, 4, 17, 0, 0)                                                       │                                                                        │
│ │       time_to = datetime.datetime(2025, 4, 16, 0, 0)                                                       │                                                                        │
│ │ total_seconds = 31449600                                                                                   │                                                                        │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                                                                        │
│                                                                                                                                                                                       │
│ /home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/core/script_runner.py:316 in run                                                                    │
│                                                                                                                                                                                       │
│   313 │   │   :param on_progress: Callback to call on every iteration                                                                                                                 │
│   314 │   │   :raises AssertionError: If the 'main' function does not return a dictionary                                                                                             │
│   315 │   │   """                                                                                                                                                                     │
│ ❱ 316 │   │   for _ in self.run_iter(on_progress=on_progress):                                                                                                                        │
│   317 │   │   │   pass                                                                                                                                                                │
│   318                                                                                                                                                                                 │
│                                                                                                                                                                                       │
│ ╭────────────────────────────────────────────────────────────────────────── locals ───────────────────────────────────────────────────────────────────────────╮                       │
│ │    _ = (OHLCV(timestamp=1713744000, open=64941.1484375, high=67232.3515625, low=64500.0, close=66819.3203125, volume=31397.994140625, extra_fields={}), {}) │                       │
│ │ self = <pynecore.core.script_runner.ScriptRunner object at 0x70774e9d5bc0>                                                                                  │                       │
│ ╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                       │
│                                                                                                                                                                                       │
│ /home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/core/script_runner.py:222 in run_iter                                                               │
│                                                                                                                                                                                       │
│   219 │   │   │   │   │   position.process_orders()                                                                                                                                   │
│   220 │   │   │   │                                                                                                                                                                   │
│   221 │   │   │   │   # Run the script                                                                                                                                                │
│ ❱ 222 │   │   │   │   res = self.script_module.main()                                                                                                                                 │
│   223 │   │   │   │                                                                                                                                                                   │
│   224 │   │   │   │   # Update plot data with the results                                                                                                                             │
│   225 │   │   │   │   if res is not None:                                                                                                                                             │
│                                                                                                                                                                                       │
│ ╭───────────────────────────────────────────────────────────────────────────────────── locals ──────────────────────────────────────────────────────────────────────────────────────╮ │
│ │    _parse_timezone = <functools._lru_cache_wrapper object at 0x70774eb52fb0>                                                                                                      │ │
│ │           barstate = <module 'pynecore.lib.barstate' from '/home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/lib/barstate.py'>                    │ │
│ │             candle = OHLCV(timestamp=1713830400, open=66819.3203125, high=67183.0078125, low=65765.8125, close=66414.0, volume=22599.900390625, extra_fields={})                  │ │
│ │       extra_fields = {'Fast EMA': NA[float], 'Slow EMA': NA[float]}                                                                                                               │ │
│ │ function_isolation = <module 'pynecore.core.function_isolation' from                                                                                                              │ │
│ │                      '/home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/core/function_isolation.py'>                                              │ │
│ │           is_strat = False                                                                                                                                                        │ │
│ │                lib = <module 'pynecore.lib' from '/home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/lib/__init__.py'>                             │ │
│ │           position = None                                                                                                                                                         │ │
│ │                res = None                                                                                                                                                         │ │
│ │               self = <pynecore.core.script_runner.ScriptRunner object at 0x70774e9d5bc0>                                                                                          │ │
│ │             string = <module 'pynecore.lib.string' from '/home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/lib/string.py'>                        │ │
│ │          trade_num = 0                                                                                                                                                            │ │
│ │     updated_candle = OHLCV(                                                                                                                                                       │ │
│ │                      │   timestamp=1713744000,                                                                                                                                    │ │
│ │                      │   open=64941.1484375,                                                                                                                                      │ │
│ │                      │   high=67232.3515625,                                                                                                                                      │ │
│ │                      │   low=64500.0,                                                                                                                                             │ │
│ │                      │   close=66819.3203125,                                                                                                                                     │ │
│ │                      │   volume=31397.994140625,                                                                                                                                  │ │
│ │                      │   extra_fields={'Fast EMA': NA[float], 'Slow EMA': NA[float]}                                                                                              │ │
│ │                      )                                                                                                                                                            │ │
│ ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                                                                                                       │
│ /home/kpob/workspace/trading/pine/workdir/scripts/demo.py:25 in main                                                                                                                  │
│                                                                                                                                                                                       │
│   22 │   A simple EMA crossover demo                                                          ╭─────── locals ────────╮                                                               │
│   23 │   """                                                                                  │ fast_length = 12      │                                                               │
│   24 │   # Calculate EMAs                                                                     │ slow_length = 26      │                                                               │
│ ❱ 25 │   fast_ema = ta.ema(src, fast_length)                                                  │         src = 66414.0 │                                                               │
│   26 │   slow_ema = ta.ema(src, slow_length)                                                  ╰───────────────────────╯                                                               │
│   27 │                                                                                                                                                                                │
│   28 │   # Plot our indicators                                                                                                                                                        │
│                                                                                                                                                                                       │
│ /home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/lib/ta.py:474 in ema                                                                                │
│                                                                                                                                                                                       │
│    471 │                                                                                        ╭──────────── locals ────────────╮                                                    │
│    472 │   # Use SMA at warming stage                                                           │   _alpha = None                │                                                    │
│    473 │   if isinstance(last_val, NA):                                                         │    alpha = 0.15384615384615385 │                                                    │
│ ❱  474 │   │   last_val = sma(source, length)                                                   │ last_val = NA[float]           │                                                    │
│    475 │   │   return cast(float | NA[float], last_val)                                         │   length = 12                  │                                                    │
│    476 │                                                                                        │   source = 66414.0             │                                                    │
│    477 │   # Warmed result                                                                      ╰────────────────────────────────╯                                                    │
│                                                                                                                                                                                       │
│ /home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/lib/ta.py:1441 in sma                                                                               │
│                                                                                                                                                                                       │
│   1438 │   :param length: The length of the moving average                                      ╭───── locals ─────╮                                                                  │
│   1439 │   :return: The Simple Moving Average (SMA)                                             │ length = 12      │                                                                  │
│   1440 │   """                                                                                  │ source = 66414.0 │                                                                  │
│ ❱ 1441 │   return lib_math.sum(source, length) / length                                         ╰──────────────────╯                                                                  │
│   1442                                                                                                                                                                                │
│   1443                                                                                                                                                                                │
│   1444 @export                                                                                                                                                                        │
│                                                                                                                                                                                       │
│ /home/kpob/.pyenv/versions/3.13.0/envs/pine/lib/python3.13/site-packages/pynecore/lib/math.py:353 in sum                                                                              │
│                                                                                                                                                                                       │
│   350 │   elif __persistent_summ_count__ == length - 1:                                        ╭───── locals ─────╮                                                                   │
│   351 │   │   __persistent_summ_count__ += 1                                                   │ length = 12      │                                                                   │
│   352 │   else:                                                                                │ source = 66414.0 │                                                                   │
│ ❱ 353 │   │   __persistent_summ_summ__ -= float(__series_summ_source__[length])                ╰──────────────────╯                                                                   │
│   354 │                                                                                                                                                                               │
│   355 │   __persistent_summ_summ__ += source                                                                                                                                          │
│   356                                                                                                                                                                                 │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: NA.__float__ returned non-float (type NA)
```

--- a/src/pynecore/cli/app.py
+++ b/src/pynecore/cli/app.py
@@ -12,7 +12,7 @@ except ImportError:
 __all__ = ["app", "app_state"]
 
 app = typer.Typer(
-    no_args_is_help=True,
+    invoke_without_command=True,
     context_settings={"help_option_names": ["-h", "--help"]},
     rich_markup_mode="rich",
 )

--- a/src/pynecore/cli/commands/__init__.py
+++ b/src/pynecore/cli/commands/__init__.py
@@ -34,10 +34,13 @@ def setup(
     """
     Pyne Command Line Interface
     """
-    if ctx.resilient_parsing or ctx.invoked_subcommand is None:
+    if ctx.resilient_parsing:
         return
-    if any(arg in ('-h', '--help') for arg in sys.argv[1:]):
-        return
+    
+    # If no subcommand is provided, show complete help like --help
+    if ctx.invoked_subcommand is None:
+        typer.echo(ctx.get_help())
+        raise typer.Exit(0)
 
     typer.echo("")
 

